### PR TITLE
fix for duplicate declarations

### LINKED ISSUE #150: Can't use same for loop var
given:

```rust
fn main(pub public_input: Field, private_input: [Field; 3]) {
    let mut sum = 0;

    for ii in 0..3 {
        sum = sum + private_input[ii];
    
    }
    for ii in 0..3 {
        sum = sum + private_input[ii];
    }

    assert_eq(sum, public_input);
}
```

it throws error at the second for loop:
```
   ╭─[for_loop:7:1]
 7 │     }
 8 │     for ii in 0..3 {
   ·         ─┬
   ·          ╰── here
 9 │         sum = sum + private_input[ii];
   ╰────
  help: the variable `ii` is declared twice
```

The cause is the `TypedFnEnv`  use var name as key to store vars, but it doesn't remove the vars that are out of scope.

### PR DESCRIPTION
**Pull request by Prabhat1308**
fixes #150 .

### CODE REVIEW

File: src/type_checker/fn_env.rs [hunk 1]
```diff
@@ -80,14 +75,12 @@ impl TypedFnEnv {
 
     /// Exits a scoped block.
     pub fn pop(&mut self) {
-        self.current_scope.checked_sub(1).expect("scope bug");
+        self.current_scope = self.current_scope.checked_sub(1).expect("scope bug");
```
Comment (mimoo) [inline src/type_checker/fn_env.rs:78]: woop! did we have a bug here o_O?

Comment (Prabhat1308) [reply src/type_checker/fn_env.rs:78]: Yes. It was not  assigning the result  after the subtraction resulting in abnormal values of `current_scope` .

File: examples/example.no [hunk 2]
```diff
@@ -1,9 +1,16 @@
 use std::crypto;
 
-fn main(pub public_input: Field, private_input: [Field; 2]) {
-    let xx = private_input[0] + private_input[1];
-    assert_eq(xx, 2);
+fn main(pub public_input: Field, private_input: [Field; 3]) {
```
Comment (katat) [inline examples/example.no:3]: please create a new test, instead of changing this existing one.

**Review by katat — COMMENTED** [review]
This is awesome!
The only comment I have is to create a new test for this case instead of modifying the `example.no`.

File: examples/dup_var.no [hunk 3]
```diff
@@ -0,0 +1,13 @@
+fn main(pub public_input: Field, private_input: [Field; 3]) {
```
Comment (katat) [inline examples/dup_var.no:1]: please add a test in the `tests/examples.rs` to point to this test code.

Comment (Prabhat1308) [reply examples/dup_var.no:1]: Added the test .

**Review by katat — APPROVED** [review]
Great, thanks!

**Pull request merged**